File: src/programmers/p1/P1_82612.java
package programmers.p1;

import java.util.stream.Stream;

//프로그래머스 위클리 챌린지
public class P1_82612 {
    public static void main(String[] args) {
        System.out.println(solution(3, 20, 4));
    }

    public static long solution(int price, int money, int count) {
        long answer = Stream.iterate(1, v -> v + 1).limit(count).mapToLong(v -> (long) v * price).sum() - money;
        if (answer > 0)
            return answer;
        return 0;
    }
}

File: src/programmers/p1/P1_82612.kt
package programmers

//프로그래머스 위클리 챌린지
fun main() {
    println(solution(3, 20, 4))
    println(solution(21, 20, 1))
}

private fun solution(price: Int, money: Int, count: Int): Long = (1..count).map { it * price.toLong() }.sum().let { if (money > it) 0 else it - money }
private fun solution2(price: Int, money: Int, count: Int): Long = (1..count).sumOf { it * price.toLong() }.let { if (money > it) 0 else it - money }